mlflow/genai/datasets/evaluation_dataset.py

```diff
@@ -1,118 +1,235 @@
+            self._mlflow_dataset._records = value
+
+    def __getattr__(self, name):
+        """
+        Dynamic attribute delegation for simple pass-through properties.
 
-    @property
-    def digest(self) -> str | None:
-        """String digest (hash) of the dataset provided by the caller that uniquely identifies"""
-        # NB: The managed Dataset entity in Agent SDK doesn't propagate the digest
-        # information. So we compute the digest of the dataframe view.
-        if self._digest is None:
-            self._digest = self._dataset.digest or compute_pandas_digest(self.to_df())
-        return self._digest
+        This handles attributes that don't require special logic and can be
+        directly delegated to the underlying dataset implementation.
+        """
+        if name.startswith("_"):
+            raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
 
-    @property
-    def name(self) -> str | None:
-        """The UC table name of the dataset."""
-        return self._dataset.name
+        if self._mlflow_dataset and hasattr(self._mlflow_dataset, name):
+            return getattr(self._mlflow_dataset, name)
+        elif self._databricks_dataset and hasattr(self._databricks_dataset, name):
+            return getattr(self._databricks_dataset, name)
 
-    @property
-    def schema(self) -> str | None:
-        """The schema of the dataset."""
-        return self._dataset.schema
+        raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
 
     @property
-    def profile(self) -> str | None:
-        """The profile of the dataset, summary statistics."""
-        return self._dataset.profile
+    def digest(self) -> str | None:
+        """String digest (hash) of the dataset provided by the caller that uniquely identifies"""
+        if self._mlflow_dataset:
+            return self._mlflow_dataset.digest
+        if self._databricks_dataset.digest is None:
+            from mlflow.data.digest_utils import compute_pandas_digest
+
+            return compute_pandas_digest(self.to_df())
+        return self._databricks_dataset.digest
 
     @property
-    def source(self) -> DatasetSource:
+    def source(self):
         """Source information for the dataset."""
+        if self._mlflow_dataset:
+            return self._mlflow_dataset.source
         return DatabricksEvaluationDatasetSource(table_name=self.name, dataset_id=self.dataset_id)
 
     @property
     def source_type(self) -> str | None:
-        """The type of the dataset source, e.g. "databricks-uc-table", "DBFS", "S3", ..."""
-        return self._dataset.source_type
+        """The type of the dataset source."""
+        if self._mlflow_dataset:
+            return self._mlflow_dataset.source._get_source_type()
+        return self._databricks_dataset.source_type
 
     @property
-    def create_time(self) -> str | None:
+    def created_time(self) -> int | str | None:
```

Comment:
can we unify the return value to `int` ?

Reply:
As this is for databricks-agents, we can't make any adjustments to the signature here in order to avoid breaking changes. This time information in databricks-agents is actually stored as a datetime string in ISO format. In MLflow, we're storing it in a more efficient way that is better for query performance. 
While I originally planned to unify these by embedding ISO -> Unix TS conversion, the chances that timezone embeddings are present risks creating a broken experience for existing users in databricks-agents.